### ISSUE TITLE
creating a palette and a canvas container

### ISSUE THREAD

**Issue by SteveAlexander** [AshesOfOwls/jquery.shapeshift#43]
I've been having a lot of fun playing with shapeshift.

I'm trying to do something with two shapeshift containers, and I can't figure out how to do it. I think it may not be possible with the current shapeshift.

The overall effect I'm going for is to have a fixed palette of four items, and be able to drag a clone of any of the items into a shapeshift container that accepts cross-drops. Upon dragging an item from the palette, it creates a new clone in the shapeshift container.

So, maybe I'd start off with my palette containing [A B C D] and my shapeshift container empty: [ . . . . . . ]

Then, as I drag, say, B across from the palette to the container, I end up with [A B C D] still in the palette, and [B. . . . . . ] in the container. I could drag another B across to make [B.B. . . . . ], or any of the other letters.

I've tried doing this by setting up two shapeshift containers, setting the "palette" one to enableDrag: true, enableCrossDrop: false, dragClone: true, deleteClone: true, but it's not working as I'd like.

Before I describe the issues I've encountered with this, can you tell me if this is a good way to approach this?

I started off thinking I'd disable drop on the "palette", not just for cross-drops, but for drops from itself. But I can't find a way to do that. Setting deleteClone sort of does it, but it still triggers a re-arrange event. I've also found that the clones behave strangely once they've been dropped onto the target container. For example, if I later drag one into a trash container, the items remain where they were, but all such items (original and clone) stop being draggable.

Any advice is most welcome! Thank you.

**Comment by SteveAlexander** [comment]
I just noticed that a clone element retains the class "ui-draggable-dragging" and "z-index: 9999" even after dragging ends. So maybe the clone needs to have some tear-down happen when dragging ends.

Also, I see from the code that the clone is made withDataAndEvents, but without the deep copy of data and events. (See http://api.jquery.com/clone/)  Maybe there's some sharing going on between a clone and an original that is making my cloned shapeshift elements work weirdly.

**Comment by AshesOfOwls** [comment]
This should be possible, I'm a little busy with my normal work right now but I will see if I can set up a demo example tonight.

**Comment by AshesOfOwls** [comment]
Yeah there is definitely some wonkeyness with the clone and the events. Still taking a look at it and will try to patch it up.

**Comment by AshesOfOwls** [comment]
This is somewhat related to #34. The cloning is fine but what needs to happen is for the item to get dragging reinitialized when it gets added to its new container. I will address this at the same time I fix that other issue.
